apis/v1alpha1/gateway_types.go

```diff
@@ -297,6 +297,35 @@ const (
 	UDPProtocolType ProtocolType = "UDP"
 )
 
+// TLSRouteOverrideType type defines the level of allowance for Routes
+// to override a specific TLS setting.
+// +kubebuilder:validation:Enum=AllowAll;DenyAll
+// +kubebuilder:default=DenyAll
+type TLSRouteOverrideType string
+
+const (
+	// AllowAll allows the parameter to be configured from all routes.
+	AllowAll TLSRouteOverrideType = "AllowAll"
+
+	// DenyAll prohibits the parameter to be configured from any route.
+	DenyAll TLSRouteOverrideType = "DenyAll"
+)
+
+// TLSOverridePolicy defines a schema for overriding TLS settings at the Route
+// level.
+type TLSOverridePolicy struct {
+	// Certificate dictates if TLS certificates can be configured
+	// via Routes. If set to 'AllowAll', a TLS certificate for a hostname
+	// defined in a Route takes precedence over the certificate defined in
+	// Gateway.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:default=DenyAll
+	Certificate TLSRouteOverrideType `json:"certificate"`
```

Comment:
@hbagdi here is an [example](https://github.com/projectcontour/contour-operator/pull/24/files) of using kubebuilder defaulting for required custom types.

Reply:
Does this need to be pointer type (optional struct)

Reply:
@bowei the field does not need to be a pointer since we're using defaulting. @hbagdi if we don't want to require that users specify a value for this field, then add the `omitempty` json tag.

Reply:
`// +optional` should also be removed since defaulting will set set the value.